graylog2-server/src/main/java/org/graylog2/messageprocessors/MessageFilterChainProcessor.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.messageprocessors;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.google.common.collect.ComparisonChain;
+import com.google.common.collect.Ordering;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.Messages;
+import org.graylog2.plugin.ServerStatus;
+import org.graylog2.plugin.filters.MessageFilter;
+import org.graylog2.plugin.messageprocessors.MessageProcessor;
+import org.graylog2.shared.buffers.processors.ProcessBufferProcessor;
+import org.graylog2.shared.journal.Journal;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+public class MessageFilterChainProcessor implements MessageProcessor {
+    private static final Logger LOG = LoggerFactory.getLogger(MessageFilterChainProcessor.class);
+
+    private final List<MessageFilter> filterRegistry;
+    private final MetricRegistry metricRegistry;
+    private final Journal journal;
+    private final ServerStatus serverStatus;
+    private final Meter filteredOutMessages;
+
+    @Inject
+    public MessageFilterChainProcessor(MetricRegistry metricRegistry,
+                                       Set<MessageFilter> filterRegistry,
+                                       Journal journal,
+                                       ServerStatus serverStatus) {
+        this.metricRegistry = metricRegistry;
+        this.journal = journal;
+        this.serverStatus = serverStatus;
+        // we need to keep this sorted properly, so that the filters run in the correct order
+        this.filterRegistry = Ordering.from(new Comparator<MessageFilter>() {
+            @Override
+            public int compare(MessageFilter filter1, MessageFilter filter2) {
+                return ComparisonChain.start()
+                        .compare(filter1.getPriority(), filter2.getPriority())
+                        .compare(filter1.getName(), filter2.getName())
+                        .result();
+            }
+        }).immutableSortedCopy(filterRegistry);
+
+        if (filterRegistry.size() == 0)
+            throw new RuntimeException("Empty filter registry!");
```

Comment:
This should be an `IllegalArgumentException` (could also use `Preconditions#checkArgument()`).